src/java.base/macosx/classes/sun/nio/fs/BsdFileStore.java

```diff
@@ -100,6 +101,10 @@ public boolean supportsFileAttributeView(Class<? extends FileAttributeView> type
             UnixPath dir = new UnixPath(file().getFileSystem(), entry().dir());
             return isExtendedAttributesEnabled(dir);
         }
+        // POSIX attributes not supported on FAT
+        if (type == PosixFileAttributeView.class &&
+            entry().fstype().equals("msdos"))
+            return false;
```

Comment:
I suspect supportsFileAttributeView(String) will need to be changed to return false if the view name is "unix".

Reply:
Yes. Changed in 0cc942c2c3d888cf90f7fcc0dd8fb6c50de616cf.